ping/ping.c

```diff
@@ -1770,12 +1770,11 @@ char *pr_raw_addr(struct ping_rts *rts, void *sa, socklen_t salen)
 char *_pr_addr(struct ping_rts *rts, void *sa, socklen_t salen, int resolve_name)
 {
 	static char buffer[4096] = "";
-	static struct sockaddr_storage last_sa;
+	static struct sockaddr_storage last_sa = {0};
 	static socklen_t last_salen = 0;
 	char name[NI_MAXHOST] = "";
 	char address[NI_MAXHOST] = "";
 
-	memset(&last_sa, 0, sizeof(last_sa));
```

Comment:
I suppose that the DNS problem was caused by this particular change. Notice that the `struct sockaddr_storage` is declared `'static`, that means that it's not allocated on the function stack, rather than it's allocated in the program data and the variable lifetime is not limited by the function scope. Hence the code that  does `= {0}` initializes the variable once at the start of the program, while the`memset()` clears it on each function call.